message/messaging_service.cc

```diff
@@ -691,22 +712,31 @@ shared_ptr<messaging_service::rpc_protocol_client_wrapper> messaging_service::ge
         remove_error_rpc_client(verb, id);
     }
 
-    auto must_encrypt = [&id, this] {
+    auto must_encrypt = [&id, &verb, this] {
         if (_cfg.encrypt == encrypt_what::none) {
             return false;
         }
         if (_cfg.encrypt == encrypt_what::all) {
             return true;
         }
 
+        // if we have dc/rack encryption but this is gossip, we should
+        // use tls anyway, to avoid having mismatched ideas on which 
+        // group we/client are in. 
+        if (verb >= messaging_verb::GOSSIP_DIGEST_SYN && verb <= messaging_verb::GOSSIP_SHUTDOWN) {
+            return true;
+        }
```

Comment:
Can this cause a cluster not to boot?

Example:

1-dc cluster with encrypt_what == dc, and misconfigured TLS.

before, we did not use TLS and everything worked.
after, we'll use misconfigured TLS and fail.

Perhaps it's unavoidable, but there should be a good explanation in the release notes.

Reply:
It might be a problem, yes. Even an update issue. Problem is, we can't make this a feature, since features are communicated via gossip. But as stated earlier, if we don't enforce TLS for gossip in this case, we cannot reach a point where we safely know all dc/rack memberships.